WordPress/Classes/ViewRelated/Post/Search/PostSearchSuggestionsService.swift

```diff
@@ -28,12 +28,16 @@ actor PostSearchSuggestionsService {
         let tokens = await [authors, tags]
         let selectedTokenIDs = Set(selectedTokens.map(\.id))
 
-        return Array(tokens
+        let output = Array(tokens
             .flatMap { $0 }
             .filter { !selectedTokenIDs.contains($0.token.id) }
             .sorted { ($0.score, $0.token.value) > ($1.score, $1.token.value) }
             .map { $0.token }
             .prefix(3))
+
+        // Remove duplicates
+        var encounteredIDs = Set<AnyHashable>()
+        return output.filter { encounteredIDs.insert($0.id).inserted }
```

Comment:
FYI there's an Array extension method called `deduplicated()`!

Reply:
Unfortunately, I could use it as is. The issue is that the elements are not Hashable, and I have to extract the ID here: `insert($0.id)`. Maybe there was a better way to design the original protocols for filters, but I decided to keep it.